x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_embeddable_factory.tsx

```diff
@@ -211,11 +216,13 @@ export const getAnomalySwimLaneEmbeddableFactory = (
           const I18nContext = i18n.Context;
           const timeBuckets = useTimeBuckets(uiSettings);
 
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
+          }
+
           useReactEmbeddableExecutionContext(
             services[0].executionContext,
-            // TODO https://github.com/elastic/kibana/issues/180055
-            // @ts-ignore
-            parentApi?.executionContext?.value ?? { name: 'dashboard' },
```

Comment:
it should be an app id, so as a temp fix before this PR, I added a hardcoded `dashboard` value

Reply:
In our case we I get `dashboards` for the executionContext with the following code:

```
const { executionContext } = useKibana().services;
const executionContextName = executionContext.get().name; // it gives dashboards
```

<img width="600" alt="Screenshot 2024-04-22 at 21 22 19" src="https://github.com/elastic/kibana/assets/2852703/2109ccce-3e92-45a9-824a-28889c7fecf1">

I searched for the use of `dashboards` and found [this](https://github.com/elastic/kibana/blame/main/src/plugins/dashboard/public/dashboard_constants.ts#L53),

```
export const DASHBOARD_APP_ID = 'dashboards';
export const LEGACY_DASHBOARD_APP_ID = 'dashboard';
```

and this [one](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ml/public/application/explorer/anomaly_context_menu.tsx#L196) 

As you said you removed the hardcoded `dashboard` value that was added as a temp fix before this PR. I don't want to be a blocker here. @ThomThomson I am just questioning if we should add:

```
 this.executionContext = {
      type: 'application',
      name: DASHBOARD_APP_ID // dashboards
      description: initialInput.title,
    };
```

instead of what is added in this PR

```
    this.executionContext = {
      type: DASHBOARD_CONTAINER_TYPE, // dashboard
      description: initialInput.title,
    };
```

Reply:
The execution context is meant to be used for telemetry. If we change the execution context name we could very well mess up the collection of certain telemetry. The key should stay exactly the same as it was.

It looks like it is currently hardcoded to `dashboard` https://github.com/elastic/kibana/blob/1e79cfc47d1d989f97b813d8ad91594a892aec75/src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts#L240C25-L240C41

@mgiota, did the code that checked for `dashboards` actually work correctly? You can always change that code to check in a better way - perhaps by checking the type of the parent instead of the execution context?

@darnautov, it looks like the Dashboard now has two execution contexts (one from create_dashboard and the other from your addition to the dashboard_container) can you consolidate these?

Reply:
@ThomThomson Yep the code that checks for `dashboards` works properly. If I change it to `dashboard` it breaks. We are probably talking about 2 different contexts (apart from what @darnautov is adding here). I checked and the one that I am getting is coming from [here](https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx#L92). 

I can describe our use case, if it helps clarify the execution context I am using. It might be different from what Dima is trying to achieve here, I just wanted to bring this into your attention.

In `SLO detail page` we have an `Error budget burn down chart` and an action to `Add to Dashboard`

<img width="1455" alt="Screenshot 2024-04-23 at 10 03 35" src="https://github.com/elastic/kibana/assets/2852703/c97663c6-6267-4b19-95b6-b05d662dd52a">

In `Dashboard` app, we render the `Error budget burn down chart`, which on click opens the SLO detail page in a `Flyout`. Within the Flyout, we disable the `Add to Dashboard` option. 

<img width="1461" alt="Screenshot 2024-04-23 at 10 05 21" src="https://github.com/elastic/kibana/assets/2852703/ff127a9b-de02-4191-8139-9b8753b05ad2">

We simply get the execution context from kibana services
`const { executionContext } = useKibana().services;`. I don't hardcode anything, I get what I receive from this call, which is following. As I posted above, it gets its value from [here](https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx#L92). :

```
{
      type: 'application',
      name:  'dashboards'
      page: 'app',
    };
```

Reply:
@ThomThomson consolidated the `executionContext` in https://github.com/elastic/kibana/pull/180904/commits/616fb15a35ba5aaa188d8c84d4fe9fd9ef95c67a

Reply:
@darnautov the consolidation looks good!

@mgiota thank you for explaining the use case. Let me look into this to see where this is coming from.